aws/src/main/java/org/apache/iceberg/aws/glue/LockManager.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.Map;
+import java.util.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * An interface for locking, used to ensure Glue catalog commit isolation.
+ */
+interface LockManager extends AutoCloseable {
+
+  /**
+   * Try to acquire a lock
+   * @param lockId lock ID
+   * @throws LockNotAcquiredException if lock is not acquired
+   */
+  void acquire(LockId lockId);
+
+  /**
+   * Release a lock
+   * @param lockId lock ID
+   * @throws IllegalArgumentException if lock entity not found or trying to unlock with a wrong owner ID
+   */
+  void release(LockId lockId);
+
+  /**
+   * Initialize lock manager from catalog properties.
+   * @param properties catalog properties
+   */
+  void initialize(Map<String, String> properties);
+
+  class LockId {
+
+    private final String entityId;
+    private final String ownerId;
+
+    LockId(String entityId, String ownerId) {
```

Comment:
Why does this include owner ID? If it identifies a lock, that shouldn't depend on owner. And if this identifies a lock, what is the entity ID?

It seems like you want the lock manager to have an ID (owner ID) and a lock to have an ID (entity ID), but this mixes them together.

Reply:
The reason I did not use an id for the lock manager is that one lock manager can acquire an entity for multiple times when it is used in multiple threads. By using the `newMetadataLocation` as the `ownerId`, we guarantee only the lock owner can unlock.

Maybe `LockId` is not a good concept to group those two concepts, I can separate them and directly use these terms in the method signature, make it like `acquire(String entityId, String ownerId)`.